### ISSUE TITLE
Atomic update to an offline configuration archive causes subsequent changes to be missed

### ISSUE THREAD

**Issue by gmckerrell** [launchdarkly/ld-relay#280]
**Describe the bug**
When operating in "offline" mode, the relay proxy watches for configuration archive changes.
The nsnotify project explicitly recommends that directories should be watched not specific files
https://pkg.go.dev/github.com/fsnotify/fsnotify#NewWatcher
> Watching individual files (rather than directories) is generally not recommended as many programs (especially editors) update files atomically: it will write to a temporary file which is then moved to to destination, overwriting the original (or some variant thereof). The watcher on the original file is now lost, as that no longer exists.

However currently the relay proxy watches the explicit archive file path.

This means that an atomic update will work exactly once; the `CHMOD` and  `REMOVED` events get signaled and the appropriate update occurs, however the removal of the file being watched means that subsequent changes are not seen.

**To reproduce**
Start up relay proxy using `offline/config.tar.gz` as the offline configuation.
```
$ cp config1.tar.gz offline/config.tar.gz.tmp
$ mv offline/config.tar.gz.tmp offline/
```
config1 details are reloaded
```
$ cp config2.tar.gz offline/config.tar.gz.tmp
$ mv offline/config.tar.gz.tmp offline/
```
config2 details are *not* reloaded

**Expected behavior**
atomic moving of configuration archive into place works

**Logs**
The following test logs indicates the failure mode
![image](https://github.com/launchdarkly/ld-relay/assets/31652935/bcd7c087-a3d1-4305-8481-cd13a3aef7c1)
```golang
func TestFileAtomicUpdatedWithValidDataAddedEnvironment(t *testing.T) {
	archiveManagerTest(t, func(filePath string) {
		writeAtomicArchive(t, filePath, false, nil, testEnv1)
	}, func(p archiveManagerTestParams) {
		require.NoError(t, p.archiveManagerError)

		p.expectEnvironmentsAdded(testEnv1)

		writeAtomicArchive(t, p.filePath, false, nil, testEnv1, testEnv2)

		p.expectEnvironmentsAdded(testEnv2)
		p.expectReloaded()

		writeAtomicArchive(t, p.filePath, false, nil, testEnv1)

		p.expectEnvironmentsDeleted(testEnv2.id())
		p.expectReloaded()
	})
}
```
**SDK version**
8.2.1

**Language version, developer tools**
using the following docker image to build and test
```
FROM golang:alpine

RUN apk add make bash gcc libc-dev
RUN mkdir /.cache && chmod a+w -R /.cache
WORKDIR /wrk
ENV CGO_ENABLED=1
```

**OS/platform**
docker image as above

**Comment by cwaldren-ld** [comment]
Hi @gmckerrell, this definitely seems like unintended behavior. Really appreciate the report & PR. 

I should have time to look at this once some pending maintenance releases are out the door.

_Filed internally as 228224._

**Comment by cwaldren-ld** [comment]
Hi @gmckerrell , this has been released in [v8.2.3](https://github.com/launchdarkly/ld-relay/releases/tag/v8.2.3) and backported to [v7.4.4](https://github.com/launchdarkly/ld-relay/releases/tag/v7.4.4). Thanks again!

I've noticed some intermittent flakiness with the offline mode integration tests in the past couple days post-merging the fix, so I'll be monitoring those.
